packages/components/src/tooltip/test/index.js

```diff
@@ -11,60 +11,55 @@ import Tooltip from '../';
  * WordPress dependencies
  */
 import { TOOLTIP_DELAY } from '../index.js';
-import { act } from '@testing-library/react';
 
 describe( 'Tooltip', () => {
 	describe( '#render()', () => {
 		it( 'should render children (abort) if multiple children passed', () => {
-			// Mount: Enzyme shallow does not support wrapping multiple nodes.
-			const wrapper = mount(
+			const { container } = render(
 				<Tooltip>
 					<div />
 					<div />
 				</Tooltip>
 			);
 
-			expect( wrapper.children() ).toHaveLength( 2 );
+			expect( container.children ).toHaveLength( 2 );
 		} );
 
 		it( 'should render children', () => {
-			const wrapper = shallow(
+			render(
 				<Tooltip position="bottom right" text="Help text">
 					<button>Hover Me!</button>
 				</Tooltip>
 			);
 
-			const button = wrapper.find( 'button' );
-			expect( wrapper.type() ).toBe( 'button' );
-			expect( button.children() ).toHaveLength( 1 );
-			expect( button.childAt( 0 ).text() ).toBe( 'Hover Me!' );
+			const button = screen.getByRole( 'button' );
+			expect( button.nodeName ).toBe( 'BUTTON' );
+			expect( button.childNodes ).toHaveLength( 1 );
+			expect( button ).toHaveTextContent( 'Hover Me!' );
 		} );
 
 		it( 'should render children with additional popover when over', () => {
-			const wrapper = shallow(
+			const { container } = render(
 				<Tooltip position="bottom right" text="Help text">
 					<button>Hover Me!</button>
 				</Tooltip>
 			);
 
-			const event = { type: 'focus', currentTarget: {} };
-			wrapper.simulate( 'focus', event );
+			const button = screen.getByRole( 'button' );
+			button.focus();
+			expect( button.nodeName ).toBe( 'BUTTON' );
+			expect( button.childNodes ).toHaveLength( 2 );
+			expect( button.childNodes[ 0 ] ).toHaveTextContent( 'Hover Me!' );
 
-			const button = wrapper.find( 'button' );
-			const popover = wrapper.find( 'ForwardRef(Popover)' );
-			expect( wrapper.type() ).toBe( 'button' );
-			expect( button.children() ).toHaveLength( 2 );
-			expect( button.childAt( 0 ).text() ).toBe( 'Hover Me!' );
-			expect( button.childAt( 1 ).name() ).toBe( 'ForwardRef(Popover)' );
-			expect( popover.prop( 'focusOnMount' ) ).toBe( false );
-			expect( popover.prop( 'position' ) ).toBe( 'bottom right' );
```

Comment:
My understanding is that we need to create a mock when testing component props in React Testing Library.
However, since it is recommended in RTL to test  the user's interaction with the screen and component-dependent testing has divergence from implementation., I removed these tests as they were not necessary.

Am I thinking right?🤔

Reply:
I'd personally suggest using mocks, and ensuring that the mocked component receives the necessary props. Of course, we need to test the user's interactions, but at the same time, we shouldn't be stepping into the tests of the other component (`Popover` in this case). Separation of concerns is important.

Reply:
I think we can avoid using mocks (see my previous comment on how I would write this test and also [this conversation](https://github.com/WordPress/gutenberg/pull/42981#discussion_r938894913))

Reply:
Fixed by cdafb54 👍